opennms-webapp/src/main/webapp/includes/grafana-box.jsp

```diff
@@ -0,0 +1,132 @@
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * OpenNMS(R) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+--%>
+
+<%--
+  This page is included by other JSPs to create a box containing a treemap/heatmap of
+  outages grouped by nodes, foreignSources or categories.
+
+  It expects that a <base> tag has been set in the including page
+  that directs all URLs to be relative to the servlet context.
+--%>
+
+<%@page language="java"
+        contentType="text/html"
+        session="true"
+        import="org.apache.commons.io.IOUtils,
+                org.apache.http.HttpHost,
+                org.apache.http.HttpResponse,
+                org.apache.http.client.methods.HttpGet,
+                org.apache.http.impl.client.DefaultHttpClient,
+                org.apache.http.util.EntityUtils" %>
+
+<%@taglib uri="http://java.sun.com/jsp/jstl/core" prefix="c" %>
+<%@taglib uri="http://java.sun.com/jsp/jstl/fmt" prefix="fmt" %>
+
+<%
+    String grafanaApiKey = System.getProperty("org.opennms.grafanaBox.apiKey", "");
+    String grafanaProtocol = System.getProperty("org.opennms.grafanaBox.protocol", "http");
+    String grafanaHostname = System.getProperty("org.opennms.grafanaBox.hostname", "localhost");
+    String grafanaTag = System.getProperty("org.opennms.grafanaBox.tag", "");
+    int grafanaPort = Integer.parseInt(System.getProperty("org.opennms.grafanaBox.port", "3000"));
+
+    String responseString = null;
+
+    if (!"".equals(grafanaApiKey)
+            && !"".equals(grafanaHostname)
+            && !"".equals(grafanaProtocol)
+            && ("http".equals(grafanaProtocol) || "https".equals(grafanaProtocol))) {
+        /**
+         * Creating the Http client instance...
+         */
+        DefaultHttpClient httpclient = new DefaultHttpClient();
+
+        HttpHost httpHost = new HttpHost(grafanaHostname, grafanaPort, grafanaProtocol);
+
+        /**
+         * ...using the search API url.
+         */
+        HttpGet httpGet = new HttpGet("/api/search/");
+        httpGet.setHeader("Authorization", "Bearer " + grafanaApiKey);
+
+        /**
+         * Invoke the request to the Grafana host.
+         */
+        HttpResponse httpResponse = httpclient.execute(httpHost, httpGet);
```

Comment:
Is there any reason were making a synchronous to the Grafana API on the server-side instead of using an AJAX call? The whole index page won't load until we get a response back from the Grafana server.